rust-runtime/aws-smithy-types/src/date_time/format.rs

```diff
@@ -35,101 +64,117 @@ fn remove_trailing_zeros(string: &mut String) {
+    pub(crate) fn format(date_time: &DateTime) -> String {
+        if date_time.subsecond_nanos == 0 {
+            format!("{}", date_time.seconds)
         } else {
-            let mut result = format!("{}.{:0>9}", instant.seconds, instant.subsecond_nanos);
+            let mut result = format!("{}.{:0>9}", date_time.seconds, date_time.subsecond_nanos);
             remove_trailing_zeros(&mut result);
             result
         }
     }
 
-    /// Parses the Smithy epoch seconds date-time format into an `Instant`.
-    pub(crate) fn parse(value: &str) -> Result<Instant, DateParseError> {
+    /// Parses the Smithy epoch seconds date-time format into a `DateTime`.
+    pub(crate) fn parse(value: &str) -> Result<DateTime, DateTimeParseError> {
         let mut parts = value.splitn(2, '.');
         let (mut whole, mut decimal) = (0i64, 0u32);
         if let Some(whole_str) = parts.next() {
-            whole = <i64>::from_str(whole_str).map_err(|_| DateParseError::IntParseError)?;
+            whole = <i64>::from_str(whole_str).map_err(|_| DateTimeParseError::IntParseError)?;
         }
         if let Some(decimal_str) = parts.next() {
             if decimal_str.starts_with('+') || decimal_str.starts_with('-') {
-                return Err(DateParseError::Invalid("invalid epoch-seconds timestamp"));
+                return Err(DateTimeParseError::Invalid(
+                    "invalid epoch-seconds timestamp".into(),
+                ));
             }
             if decimal_str.len() > 9 {
-                return Err(DateParseError::Invalid("decimal is longer than 9 digits"));
+                return Err(DateTimeParseError::Invalid(
+                    "decimal is longer than 9 digits".into(),
+                ));
             }
             let missing_places = 9 - decimal_str.len() as isize;
-            decimal = <u32>::from_str(decimal_str).map_err(|_| DateParseError::IntParseError)?;
+            decimal =
+                <u32>::from_str(decimal_str).map_err(|_| DateTimeParseError::IntParseError)?;
             for _ in 0..missing_places {
                 decimal *= 10;
             }
         }
-        Ok(Instant::from_secs_and_nanos(whole, decimal))
+        Ok(DateTime::from_secs_and_nanos(whole, decimal))
     }
 }
 
 pub(crate) mod http_date {
     use super::remove_trailing_zeros;
-    use crate::instant::format::{DateParseError, NANOS_PER_SECOND};
-    use crate::Instant;
-    use chrono::{Datelike, NaiveDate, NaiveDateTime, NaiveTime, Timelike, Weekday};
+    use crate::date_time::format::{DateTimeFormatError, DateTimeParseError, NANOS_PER_SECOND};
+    use crate::DateTime;
     use std::str::FromStr;
+    use time::{Date, Month, OffsetDateTime, PrimitiveDateTime, Time, UtcOffset, Weekday};
 
     // This code is taken from https://github.com/pyfisch/httpdate and modified under an
     // Apache 2.0 License. Modifications:
     // - Removed use of unsafe
     // - Add serialization and deserialization of subsecond nanos
```

Comment:
My first reaction upon seeing this code was "why aren't we using `httpdate`"? These strike me as valuable additions to contribute upstream. Can we? Also, can't we simply depend on `httpdate` and wrap it in a newtype to provide our additional functionality?

---

Also, I'm not a lawyer, but I thought that if one copied MIT-licensed code, a copy of the MIT license has to be included with it.

Reply:
`    // - Add serialization and deserialization of subsecond nanos` this is outside of the HTTP date spec (as in, "wrong" if you're actually parsing and serializing http dates.